docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,806 @@
+
+```
+DEBUG Bootstrap status: complete
+INFO It is now safe to remove the bootstrap resources
+```
+
+### Access the OpenShift API
+
+You can use the `oc` or `kubectl` commands to talk to the OpenShift API. The admin credentials are in `auth/kubeconfig`:
+
+```sh
+$ export KUBECONFIG="$PWD/auth/kubeconfig"
+$ oc get nodes
+$ oc get pods -A
+```
+
+**NOTE**: Only the API will be up at this point. The OpenShift UI will run on the compute nodes.
+
+
+### Delete the Bootstrap Resources
+
+You can now safely delete the bootstrap port, server and floating IP address:
+
+```sh
+$ openstack server delete openshift-bootstrap
+$ openstack port delete openshift-qlvwv-bootstrap-port
+$ openstack floating ip delete 203.0.113.24
+```
+
+If you hadn't done so already, you should also disable the bootstrap Ignition URL:
+
+https://example.com/bootstrap.ign
+
+
+## Compute Nodes
+
+This process is similar to the masters, but the workers need to be approved before they're allowed to join the cluster.
+
+We will create three worker nodes here.
+
+**NOTE**: some of the default pods (e.g. the `openshift-router`) require at least two nodes so that is the effective minimum.
+
+The workers need no ignition override -- we can pass the unmodified `worker.ign` as their user data:
+
+```sh
+$ for index in $(seq 0 2); do
+> openstack port create --network openshift-qlvwv-network --security-group openshift-qlvwv-worker --allowed-address ip-address=192.0.2.5 --allowed-address ip-address=192.0.2.6 --allowed-address ip-address=192.0.2.7 "openshift-qlvwv-worker-port-$index"
+> openstack server create --image rhcos --flavor m1.large  --user-data "worker.ign" --port "openshift-qlvwv-worker-port-$index" "openshift-qlvwv-worker-$index"
+> done
+```
+
+
+
+### Approve the worker CSRs (check they're the nodes you've created)
+
+Even after they've booted up, the workers will not show up in `oc get nodes`.
+
+Instead, they will create certificate signing requests (CSRs) which need to be approved. You can watch for the CSRs here:
+
+```sh
+$ watch oc get csr -A
+```
+
+Eventually, you should see `Pending` entries looking like this
+
+```
+csr-cpnwz   2m17s   system:serviceaccount:openshift-machine-config-operator:node-bootstrapper   Pending
+```
+
+You should inspect each pending CSR and verify that it comes from a node you recognise:
+
+```
+$ oc describe csr csr-lxvbt
+Name:               csr-lxvbt
+Labels:             <none>
+Annotations:        <none>
+CreationTimestamp:  Fri, 18 Oct 2019 16:28:53 +0200
+Requesting User:    system:serviceaccount:openshift-machine-config-operator:node-bootstrapper
+Status:             Pending
+Subject:
+         Common Name:    system:node:openshift-qlvwv-worker-0
+         Serial Number:
+         Organization:   system:nodes
+Events:  <none>
+```
+
+If it does (this one is for `openshift-qlvwv-worker-0` which we've created earlier), you can approve it:
+
+```sh
+$ oc adm certificate approve csr-cpnwz
```

Comment:
For doc consistency, the previous block does an oc describe on csr-lxvbt, but this approve is for csr-cpnwz.  Is that correct?

Reply:
Ooops! Fixed in the next commit. Thanks.